tests/integration/test_core.py

```diff
@@ -576,7 +577,29 @@ async def test_loki_integration(
     assert: loki joins relation successfully, logs are being output to container and to files for
         loki to scrape.
     """
-    await model.wait_for_idle(apps=[application_name, loki.name], status="active")
+
+    async def wait_loki_unit_agent_idle():
```

Comment:
I'm curious: Why is this needed?

Reply:
The agent can still be executing when the `model.wait_for_idle` returns (this function isn't quite 100%), and when the assertions (querying from the loki api) are executed, it returns the wrong result which doesn't contain the access logs and error logs. It's a messy workaround for having no `agent.wait_for_idle()` or `unit.wait_for_idle().
I've tested the version with and without on the actions (the tests always passes on my local ;_;), the pass rate with this workaround is much higher than the ones without. :)
Thanks for the review!

Reply:
It's a nice workaround :) What do you think about adding a small comment regarding that? Usually, we see only the model.wait_for_idle.

Reply:
Refactored to a helper with the description in the docstring! Thanks for the suggestion :)